browser/ts/index.ts

```diff
@@ -761,6 +786,66 @@ export class Mesh {
         return wrapperStatsToStats(wrapperStats);
     }
 
+    /**
+     * Get all 0x signed orders currently stored in the Mesh node
+     * @param perPage number of signedOrders to fetch per paginated request
+     * @returns the snapshotID, snapshotTimestamp and all orders, their hashes and fillableTakerAssetAmounts
+     */
+    public async getOrdersAsync(perPage: number = 200): Promise<GetOrdersResponse> {
+        await waitForLoadAsync();
+        if (this._wrapper === undefined) {
+            // If this is called after startAsync, this._wrapper is always
+            // defined. This check is here just in case and satisfies the
+            // compiler.
+            return Promise.reject(new Error('Mesh is still loading. Try again soon.'));
+        }
+
+        let snapshotID = ''; // New snapshot
+
+        // TODO(albrow): De-dupe this code with the method by the same name
+        // in the TypeScript RPC client.
+        let page = 0;
+        let getOrdersResponse = await this.getOrdersForPageAsync(page, perPage, snapshotID);
+        snapshotID = getOrdersResponse.snapshotID;
+        let ordersInfos = getOrdersResponse.ordersInfos;
+
+        let allOrderInfos: OrderInfo[] = [];
+
+        do {
+            allOrderInfos = [...allOrderInfos, ...ordersInfos];
+            page++;
+            getOrdersResponse = await this.getOrdersForPageAsync(page, perPage, snapshotID);
+            ordersInfos = getOrdersResponse.ordersInfos;
+        } while (ordersInfos.length > 0);
```

Comment:
Just out of curiosity, is there a reason why we're using a `do..while` loop here?

Reply:
You'll have to ask @fabioberger. This is copied over from the TypeScript RPC client since we don't have any way to share code between them right now.

Reply:
@jalextowle we need to do an initiate request and then feed the `snapshotID` returned into subsequent requests. Just made sense at the time I wrote it. Any issues with it?

Reply:
Nope. I like `do..while` loops, but I almost never see them used. I just wondered if there was a particular reason that we used them. It sounds like they save a "bootstrapping" statement before a for or while loop.

Thanks for answering 😄